### PR TITLE
room setup

### PR DESCRIPTION
**Pull request by r-ismer**
The purpose of the pr is to setup the local database to store the user in it when offline

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/blindly/localDB/localDBTest.kt [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        .setPassions(listOf("wine"))
+        .setAgeRange(listOf(20, 30))
+        .setRadius(50)
+        .setLikes(listOf("bob"))
+        .setMatches(listOf("bob"))
+        .setRecordingPath("path")
+        .setDescription("description")
+        .build()
+
+    /*
```
Comment (cgntnr) [inline app/src/androidTest/java/ch/epfl/sdp/blindly/localDB/localDBTest.kt:36]: We can remove this if it's not used I guess.

File: app/src/main/java/ch/epfl/sdp/blindly/localDB/UserConverter.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        return listOf(values[0].toDouble(), values[1].toDouble())
+    }
+
+    @TypeConverter
+    fun fromStringListToString(sl: List<String>): String {
+        val len = sl.size
+        val str = StringBuilder()
+        for(i in 0 until len) {
+            str.append(sl[i])
+            if (i != len - 1) {
```
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/localDB/UserConverter.kt:29]: Not very important but you can use TextUtils.join for this kind of things. 

```join(CharSequence delimiter, Iterable tokens)```

Returns a string containing the tokens joined by delimiters.

File: app/src/main/java/ch/epfl/sdp/blindly/localDB/UserDAO.kt [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+     */
+    @Query("SELECT agerange FROM userentity WHERE uid = :id")
+    fun getUserAgeRangeValue(id: String): String?
+
+    fun getUserAgeRange(id: String): List<Int> {
+        val ageRange = getUserAgeRangeValue(id)?.split(",")
+        if(ageRange != null) {
+            return listOf(ageRange[0].toInt(), ageRange[1].toInt())
+        }
+        return listOf(18, 99)
```
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/localDB/UserDAO.kt:202]: Here as well, MIN_AGE and MAX_AGE would be nicer.

**Review by cgntnr — APPROVED** [review]
Couple of comments added, otherwise all good and mostly tested. I don't see why you pushed changes in .idea/codeStyles/Project.xml though.

File: app/src/main/java/ch/epfl/sdp/blindly/localDB/UserDAO.kt (hunk 3 above)
Comment (r-ismer) [reply app/src/main/java/ch/epfl/sdp/blindly/localDB/UserDAO.kt]: done !

File: app/src/main/java/ch/epfl/sdp/blindly/localDB/UserDAO.kt [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    @Query("SELECT location FROM userentity WHERE uid = :id")
+    fun getUserLocationValue(id: String): String?
+
+
+    fun getUserLocation(id: String): List<Double> {
+        val loc = getUserLocationValue(id)?.split(",")
+        if(loc != null) {
+            return listOf(loc[0].toDouble(), loc[1].toDouble())
+        }
+        return listOf(22.5, 6.5)
```
[the remark this answers is not in the document]
Comment (r-ismer) [reply app/src/main/java/ch/epfl/sdp/blindly/localDB/UserDAO.kt]: done !

File: app/src/main/java/ch/epfl/sdp/blindly/localDB/UserConverter.kt (hunk 2 above)
Comment (r-ismer) [reply app/src/main/java/ch/epfl/sdp/blindly/localDB/UserConverter.kt:29]: tried it but can't test it with it

**Review by Lollipoke — APPROVED** [review]
I added a few comment, mostly typos, and maybe if you can check the String.join() thing it could be cool :) otherwise you did a great job !

File: app/src/main/java/ch/epfl/sdp/blindly/localDB/UserConverter.kt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                str.append(",")
+            }
+        }
+        return str.toString()
+    }
```
[the remark this answers is not in the document]
Comment (r-ismer) [reply app/src/main/java/ch/epfl/sdp/blindly/localDB/UserConverter.kt:34]: As i said to cagin I have to mack this function to ba able to pass some test on this class.

**Pull request merged**